src/care/PluginData.h

```diff
@@ -0,0 +1,64 @@
+#ifndef _CARE_PluginData_H_
+#define _CARE_PluginData_H_
+
+#include "chai/PointerRecord.hpp"
+#include "care/config.h"
+#include <vector>
+#include <functional>
+#include <unordered_map>
+
+namespace chai {
+   struct PointerRecord;
+}
+
+namespace care{
+
+	class PluginData	{
+		public:
+			PluginData() = default;
+
+         static void setFileName(const char * name);
+
+         static void setLineNumber(int num);	
+
+         static const char * getFileName();
+         
+         static int getLineNumber();   
+
+         static void setParallelContext(bool isParallel);
+
+         static bool isParallelContext();
+
+         static bool post_parallel_forall_action_registered(void * key); 
+
+         static std::unordered_map<void *, std::function<void(chai::ExecutionSpace, const char *, int)>> get_post_parallel_forall_actions();
```

Comment:
Add a comment here, since a map with a key of a void* is a bit unusual. You might also consider having something like "using ActionMap = std::unordered_map<void *, std::function<void(chai::ExecutionSpace, const char *, int)>>" or whatever since you use it in a few places

Reply:
I think this was added by Peter, so we should have him add the comment. I'm not even sure what this is used for, but I think it's something to do with race condition checking.

The alias is a good idea, though.

Reply:
just added the alias!